dotCMS/src/main/java/com/dotcms/metrics/RequestTracker.java

```diff
@@ -0,0 +1,278 @@
+package com.dotcms.metrics;
+
+import com.dotmarketing.util.Logger;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.atomic.AtomicLong;
+
+/**
+ * Unified request tracking utility that provides metrics collection and
+ * shutdown coordination for HTTP requests.
+ * 
+ * This class is thread-safe and designed to be used by both the RequestTrackingFilter
+ * and the HttpRequestMetrics binder to avoid duplication and ensure consistency.
+ * 
+ * Key responsibilities:
+ * - Track active request count for graceful shutdown
+ * - Collect HTTP status code distributions 
+ * - Track endpoint-specific request counts
+ * - Measure request durations
+ * - Support both metrics collection and shutdown coordination
+ * 
+ * @author dotCMS Team
+ */
+public class RequestTracker {
```

Comment:
RequestMetricsTracker

Reply:
Same as above.   The RequestTracker can and is being used by more than the Metrics, and is now used by request draining in shutdown also.   Any code that wants to know how many requests are currently active can use this and not just metrics.